test/TesterInternal/StorageTests/PersistenceGrainTests.cs

```diff
@@ -325,6 +325,26 @@ public void Persistence_Silo_StorageProvider_Name_Missing()
         }
 
         [Fact, TestCategory("Functional"), TestCategory("Persistence")]
+        public async Task Persistence_Read_Write_Struct_State()
+        {
+            const string providerName = "test1";
+            Guid id = Guid.NewGuid();
+            var grain = GrainClient.GrainFactory.GetGrain<IValueTypeTestGrain>(id);
+
+            var initial = await grain.GetStateData();
+            Assert.AreEqual(0, initial);
+
+            await grain.SetStateData(long.MaxValue);
+
+            MockStorageProvider storageProvider = FindStorageProviderInUse(providerName);
+
+            Assert.AreEqual(2, storageProvider.ReadCount, "StorageProvider #Reads");
+            Assert.AreEqual(1, storageProvider.WriteCount, "StorageProvider #Writes");
```

Comment:
If we weren't to use these methods (which are not really required to assert that value types works), you can move this test to the Tester project.
We should slowly stop using TesterInternal by default if possible, unless it's really testing something very internal to Orleans.